File: Tests/Bridging/Infrastructure.m
@import Foundation;
@import PromiseKit;
#import "Infrastructure.h"

AnyPromise *PMKDummyAnyPromise_YES() {
    return [AnyPromise promiseWithValue:@YES];
}

AnyPromise *PMKDummyAnyPromise_Manifold() {
    return [AnyPromise promiseWithValue:PMKManifold(@YES, @NO, @NO)];
}

AnyPromise *PMKDummyAnyPromise_Error() {
    return [AnyPromise promiseWithValue:[NSError errorWithDomain:@"a" code:1 userInfo:nil]];
}

@implementation PromiseBridgeHelper (objc)

- (AnyPromise *)bridge2 {
    return [AnyPromise promiseWithResolverBlock:^(PMKResolver resolve) {
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            resolve(@123);
        });
    }];
}

@end

#import "PMKBridgeTests-Swift.h"

AnyPromise *testCase626() {
    return PMKWhen(@[[TestPromise626 promise], [TestPromise626 promise]]).then(^(id value){
        NSLog(@"Success: %@", value);
    }).catch(^(NSError *error) {
        NSLog(@"Error: %@", error);
        @throw error;
    });
}

File: Tests/Bridging/Infrastructure.swift
import PromiseKit

// for BridgingTests.m
@objc(PMKPromiseBridgeHelper) class PromiseBridgeHelper: NSObject {
    @objc func bridge1() -> AnyPromise {
        let p = after(.milliseconds(10))
        return AnyPromise(p)
    }
}

enum MyError: Error {
    case PromiseError
}

@objc class TestPromise626: NSObject {

    @objc class func promise() -> AnyPromise {
        let promise: Promise<String> = Promise { seal in
            seal.reject(MyError.PromiseError)
        }

        return AnyPromise(promise)
    }
}
